src/System.Runtime.Caching/tests/System.Runtime.Caching/HostFileChangeMonitorTest.cs

```diff
@@ -0,0 +1,318 @@
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Caching;
+using System.Runtime.Caching.Hosting;
+using System.Text;
+
+using Xunit;
+using MonoTests.Common;
+
+namespace MonoTests.System.Runtime.Caching
+{
+    public class HostFileChangeMonitorTest
+    {
+        [Fact]
+        public void Constructor_Exceptions()
+        {
+            string relPath = Path.Combine("relative", "file", "path");
+            var paths = new List<string> {
+                relPath
+            };
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                new HostFileChangeMonitor(paths);
+            });
+
+            paths.Clear();
+            paths.Add(null);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                new HostFileChangeMonitor(paths);
+            });
+
+            paths.Clear();
+            paths.Add(String.Empty);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                new HostFileChangeMonitor(paths);
+            });
+
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                new HostFileChangeMonitor(null);
+            });
+
+            paths.Clear();
+            Assert.Throws<ArgumentException>(() =>
+            {
+                new HostFileChangeMonitor(paths);
+            });
+        }
+
+        [Fact]
+        private static void Constructor_MissingFiles_Handler()
+        {
+            HostFileChangeMonitor monitor;
+            PlatformID pid = Environment.OSVersion.Platform;
+            bool runningOnWindows = ((int)pid != 128 && pid != PlatformID.Unix && pid != PlatformID.MacOSX);
+            string missingFile = Path.Combine("missing", "file", "path");
+
+            if (runningOnWindows)
+                missingFile = "c:\\" + missingFile;
+            else
+                missingFile = "/" + missingFile;
```

Comment:
Does it need to be a full path?  Could we just use something like:
```C#
string missingFile = Path.GetFullPath(Guid.NewGuid().ToString("N"));
```
and then not need to worry about cross-platform differences in paths?

Reply:
good idea to use Path.GetFullPath - I'll change the test.